clusterprovider/kind/kind.go

```diff
@@ -163,6 +161,7 @@ func (k *KindWatcher) ResultChan() <-chan clusterprovider.WatchEvent {
 						logger.Info("Detected a new cluster")
 						lcInfo, err := k.provider.Get(ctx, name)
 						if err != nil {
+							logger.Info("Kind cluster is not ready. Retrying.", "cluster", name)
```

Comment:
We need to decide on one style of logging.  I started using the style you previously used in the reconcilers `c.logger.V(2).Error(err, "failed to get the provider resource", newCluster.Name)` but I am up to whatever you prefer as long as we're consistent :)